src/runmode-dpdk.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/** \file
+ *
+ *  \author Lukas Sismis <lukas.sismis@gmail.com>
+ */
+
+#ifndef __RUNMODE_DPDK_H__
+#define __RUNMODE_DPDK_H__
+
+typedef struct DPDKIfaceConfigAttributes_ {
```

Comment:
Can we avoid allocations for each config setting?

Reply:
this structure does not use dynamic allocations, it only serves to access YAML fields of DPDK config. The structure is defined in src/runmode-dpdk.c on L99 and uses static strings. It is only defined once in the whole application. I didn't want to use static strings of the configuration options directly in the code but instead, I wanted to centralize it into one structure.